src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AttributeCollection.cs

```diff
@@ -0,0 +1,392 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Collections;
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA2112:SecuredTypesShouldNotExposeFields", Scope = "type", Target = "System.ComponentModel.AttributeCollection")]
+
+namespace System.ComponentModel
+{
+    /// <devdoc>
+    ///     Represents a collection of attributes.
+    /// </devdoc>
+    public class AttributeCollection : ICollection
+    {
+        /// <devdoc>
+        ///     An empty AttributeCollection that can used instead of creating a new one.
+        /// </devdoc>
+        public static readonly AttributeCollection Empty = new AttributeCollection(null);
+        private static Hashtable s_defaultAttributes;
```

Comment:
There are a lot of instances of non-generic collections. I created task #8359 to do that work.